Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -804,48 +804,210 @@ Another compliant solution is to define the API level to 17 (JELLY_BEAN_MR1) and
+```
+Now in another class, you can read/write the object using an `ObjectInputStream`/`ObjectOutputStream`.
+
+##### JSON
+
+There are various ways to serialize the contents of an object to JSON. Android comes with the `JSONObject` and `JSONArray` classes. Next there is a wide veriety of libraries which can be used, such as: GSON<sup>[2]</sup>, Jackson<sup>[3]</sup> and others. They mostly differ in whether they use reflection to compose the object, whether they support annotations and the amount of memory they use. Note that almost all the JSON representations are String based and therefore immutable. This means that any secret stored in JSON will be harder to remove from memory. 
 The JSON itself can be stored somewhere (E.g. (NoSQL) database or a file). You just need to make sure that any JSON that contains secrets has been appropriately protected (e.g. encrypted/HMACed). See the storage chapter for more details.
+Here is a simple example of how JSON can be written and read using GSON from the GSON User Guide. In this sample, the contents of an instance of the `BagOfPrimitives` is serialized into JSON:
+
+```java
+class BagOfPrimitives {
+  private int value1 = 1;
+  private String value2 = "abc";
+  private transient int value3 = 3;
+  BagOfPrimitives() {
+    // no-args constructor
+  }
+}
+
+// Serialization
+BagOfPrimitives obj = new BagOfPrimitives();
+Gson gson = new Gson();
+String json = gson.toJson(obj);  
+
+// ==> json is {"value1":1,"value2":"abc"}
+
+```
 
-##### ORM
 
-Object-Relational Mapping (ORM) is used to store the contents of an object directly into a database. Libraries like OrmLite<sup>[4]</sup>, SugarORM<sup>[5]</sup>, GreenDAO<sup>[6]</sup> and ActiveAndroid<sup>[7]</sup> use a SQLite database to store the data in. Realm <sup>[8]</sup>, another library, uses its own database to store the contents of a class. 
+##### ORM
+There are libraries that provide the functionality to store the contents of an object directly into a database and then instantiate the objects based on the database content again. This is called Object-Relational Mapping (ORM). 
+There are libraries that use SQLite as a database, such as: OrmLite<sup>[4]</sup>, SugarORM<sup>[5]</sup>, GreenDAO<sup>[6]</sup> and ActiveAndroid<sup>[7]</sup> use a SQLite database to store the data in. Realm <sup>[8]</sup> on the other hand, uses its own database to store the contents of a class. 
 The amount of protection that ORM can provide mostly relies on whether the database is encrypted. See the storage chapter for more details.
+A nice example of ORMLite can be found at <sup>[9]</sup>.
```

Comment:
I don't think this reference works like this. The superscripted number usually is put on a word, like this:

> A nice example of ORMLite is HelloAndroid<sup>[9]</sup>.

Or maybe the reference shouldn't be superscripted:

> A nice example of ORMLite can be found at [9].